packages/storybook/src/schematics/migrate-defaults-5-to-6/migrate-defaults-5-to-6.ts

```diff
@@ -0,0 +1,167 @@
+          schema.uiFramework
+        )
+      : migrateAllStorybookInstances(schema.keepOld),
+    migrateRootLevelStorybookInstance(),
+    upgradeStorybookPackagesInPackageJson(),
+  ]);
+}
+
+export function migrateAllStorybookInstances(keepOld: boolean): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.debug('adding .storybook folder to lib');
+    const workspaceJson = readWorkspaceJson();
+    const projects: string[] = Object.keys(workspaceJson.projects);
+    const projectsThatHaveStorybookConfiguration: {
+      name: string;
+      uiFramework: string;
+      configFolder: string;
+    }[] = projects.map((project: string) => {
+      if (
+        workspaceJson.projects[project].targets &&
+        workspaceJson.projects[project].targets.storybook
+      )
+        return {
+          name: project,
+          uiFramework:
+            workspaceJson.projects[project].targets.storybook.options
+              .uiFramework,
+          configFolder:
+            workspaceJson.projects[project].targets.storybook.options.config
+              .configFolder,
+        };
+    });
+
+    return chain(
+      projectsThatHaveStorybookConfiguration
+        .filter((projectWithStorybook) => projectWithStorybook)
+        .map((projectWithStorybook) => {
+          return chain([
+            migrateStorybookInstance(
+              keepOld,
+              projectWithStorybook.name,
+              projectWithStorybook.uiFramework,
+              projectWithStorybook.configFolder
+            ),
+          ]);
+        })
+    );
+  };
+}
+
+export function migrateStorybookInstance(
+  keepOld: boolean,
+  projectName: string,
+  uiFramework: string,
+  configFolder?: string
+): Rule {
+  return chain([
+    migrateProjectLevelStorybookInstance(
+      keepOld,
+      projectName,
+      uiFramework,
+      configFolder
+    ),
+  ]);
+}
+
+function upgradeStorybookPackagesInPackageJson(): Rule {
+  // TODO Implement this
+
+  /**
+   * Should upgrade all @storybook/* packages in package.json
+   */
+  return (tree: Tree, context: SchematicContext) => {};
+}
+
+function deleteOldFiles(configFolderDir: string): Rule {
+  // TODO Implement this
+  return (tree: Tree, context: SchematicContext) => {};
+}
+
+function moveOldFiles(configFolderDir: string): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    return chain([move(configFolderDir, `${configFolderDir}_old`)]);
+  };
+}
+function migrateProjectLevelStorybookInstance(
+  keepOld: boolean,
+  projectName: string,
+  uiFramework: string,
+  configFolder?: string
+): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const projectConfig = getProjectConfig(tree, projectName);
+    const { projectType } = getProjectConfig(tree, projectName);
+    const projectDirectory = projectDir(projectType);
+    return chain([
+      keepOld
+        ? moveOldFiles(
+            configFolder ? configFolder : `${projectConfig.root}/.storybook`
```

Comment:
so if they don't have a configFolder we assume all the config are in the `../.storybook` folder?

would they not always need to have a `configFolder` specified in `angular.json` for our storybook builders to work?

Reply:
Yes, indeed. I changed this part. I had omitted something. I am still checking for existence, though! I know I should not, but I feel safer that way.